### ISSUE TITLE
SpectralNorm implementation

### ISSUE THREAD

**Issue by allhailjustice** [tensorflow/addons#2414]
I am reading the source code of the spectral normalization's implementation. I think it is not consistent with its original paper's algorithm. I am not an expert on tensorflow so please point out if I am wrong.

The paper's algorithm indicates updating w according to df(w/sigma)/dw. However, the tfa's implementation directly assigns w/sigma to w (as w') and then updates w' according to df(w')/dw'. In my experiment, this implementation leads to exploding loss.

Does anyone have an idea on this? Thanks

**Comment by bhack** [comment]
There is a WIP refactoring can you take a look at  https://github.com/tensorflow/addons/pull/2411?

**Comment by korneelvdbroek** [comment]
I agree with @allhailjustice 
Here is the unit test for the differentiation which fails for the current implementation https://colab.research.google.com/drive/137attP3CTBiojPvCdgZotzEwZ1qhnwmp?usp=sharing

**Comment by milmor** [comment]
What's the best way to fix TF spectral norm? It's been more than a year, and this layer still doesn't work.

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
